control/timeresp.py

```diff
@@ -1869,8 +1885,16 @@ def impulse_response(sys, T=None, X0=0., input=None, output=None, T_num=None,
     # Figure out if the system is SISO or not
     issiso = sys.issiso() or (input is not None and output is not None)
 
+    # Select only the given input and output, if any
+    input_labels = sys.input_labels if input is None \
+        else sys.input_labels[:1]
+    output_labels = sys.output_labels if output is None \
+        else sys.output_labels[:1]
```

Comment:
Now I'm thinking if it should be
```suggestion
        else sys.output_labels[output]
```

... I'll add a test case to check it